packages/vue-app/template/App.js

```diff
@@ -182,10 +182,25 @@ export default {
         if (page.$options.fetch && page.$options.fetch.length) {
           p.push(promisify(page.$options.fetch, this.context))
         }
-        if (page.$fetch) {
-          p.push(page.$fetch())
-        } else {
-          // Get all component instance to call $fetch
+        // get layout component
+        const layout = this.$children.find((component) => component._name?.toLowerCase().includes("layouts"))
```

Comment:
Refactor suggest: Maybe we can use `let root = page` and change it to parent layout instance if exists?

Reply:
I think this also makes sense to only crawl everything once, also keeping `getChildrenComponentInstancesUsingFetch` simple. What was your opinion on that @ms-fadaei?